src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const finalStates = [];
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  for (const step in actions) {
```

Comment:
Consider using a `for...of` loop instead of `for...in` for iterating over the `actions` array. The `for...in` loop is generally used for iterating over object properties and can lead to unexpected behavior when used with arrays.